### ISSUE TITLE
Image: file length does not match

### ISSUE THREAD

**Issue by emauro** [Rhymen/go-whatsapp#478]
In my app this error message appeared when a group of four images were received. The app could only download the first one. The remaining three resulted in the error message. That's odd. What might be happened? Could the error be due the images appear grouped in the device? I tried to reproduce the problem to no avail.

I suggest adding to the error message the received size and the expected one.

**Comment by SchulteMK** [comment]
I changed the error message, to be more specific. Can you please retry and give us some feedback on the problem?

**Comment by rhsobr** [comment]
It looks like that even the official API has this issue.
I know that exists an opened ticket with same subject on their hands.

**Comment by Valdenirmezadri** [comment]
Hello, since yesterday I have faced this same problem

**Comment by Valdenirmezadri** [comment]
I changed the library to ignore the size, the image arrives and is correct, I'll leave it at that for now

**Comment by fredjsf** [comment]
I have same problem.

**Comment by emauro** [comment]
It seems to happen very sporadically. Today it happened for just one device for some images sent just by one number.

**Comment by emauro** [comment]
> I changed the library to ignore the length, the image arrives and is correct, I'll leave it at that for now
> as I don't know if this will be corrected, I'll post only the link that I changed.

It solves the problem and indeed the image is correct. But I am not sure if the image will be always correct. I am pretty sure that just one byte is being added somewhere (not by go-whatsapp).

**Comment by Valdenirmezadri** [comment]
> It seems to happen very sporadically. Today it happened for just one device for some images sent just by one number.


yes, it is not all images that occur the problem, I realized that it occurs more when it is an image forwarded from another whatsapp.
I have an image with this problem, I can send you if you want to test

**Comment by Valdenirmezadri** [comment]
> > I changed the library to ignore the length, the image arrives and is correct, I'll leave it at that for now
> > as I don't know if this will be corrected, I'll post only the link that I changed.
> 
> It solves the problem and indeed the image is correct. But I am not sure if the image will be always correct. I am pretty sure that just one byte is being added somewhere (not by go-whatsapp).

What i can tell you here, is that all the images that occurred this error were photos taken by the cell phone camera and forwarded to WhatsApp
